src/test/java/com/nvidia/spark/rapids/jni/HashTest.java

```diff
@@ -510,4 +510,144 @@ void testHiveHashMixed() {
+            "doing the hive hashing"),
+             Collections.singletonList(""),
+             null);
+         ColumnVector stringResult = Hash.hiveHash(new ColumnView[]{stringListCV});
+         ColumnVector stringExpected = ColumnVector.fromInts(97, 63736, -96263528, 2112075710, 0, 0);
+         ColumnVector intListCV = ColumnVector.fromLists(
+             new ListType(true, new BasicType(true, DType.INT32)),
+             Collections.singletonList(null),
+             Arrays.asList(0, -2, 3),
+             Collections.singletonList(Integer.MAX_VALUE),
+             Arrays.asList(5, -6, null),
+             Collections.singletonList(Integer.MIN_VALUE),
+             null);
+         ColumnVector intExpected = ColumnVector.fromInts(0, -59, 2147483647, 4619, -2147483648, 0);
+         ColumnVector intResult = Hash.hiveHash(new ColumnVector[]{intListCV});) {
+      assertColumnsAreEqual(stringExpected, stringResult);
+      assertColumnsAreEqual(intExpected, intResult);
+    }
+  }
+
+  @Test
+  void testHiveHashNestedLists() {
+    try (ColumnVector nestedStringListCV = ColumnVector.fromLists(
+            new ListType(true, new ListType(true, new BasicType(true, DType.STRING))),
+            Arrays.asList(null, Arrays.asList("a", null)),
+            Arrays.asList(Arrays.asList("B\n", "")),
+            Arrays.asList(Collections.singletonList("dE\"\u0100\t\u0101"), Collections.singletonList(" \ud720\ud721")),
+            Arrays.asList(Collections.singletonList("This is a long string (greater than 128 bytes/char string) case to test this " +
+              "hash function. Just want an abnormal case here to see if any error may happen when" +
+              "doing the hive hashing")),
+            Arrays.asList(Collections.singletonList(""), null),
+            null);
+         ColumnVector stringExpected = ColumnVector.fromInts(3007, 63736, -96263528, 2112075710, 0, 0);
+         ColumnVector stringResult = Hash.hiveHash(new ColumnView[]{nestedStringListCV});
+         ColumnVector nestedIntListCV = ColumnVector.fromLists(
+             new ListType(true, new ListType(true, new BasicType(true, DType.INT32))),
+             Arrays.asList(Arrays.asList(null, null), null),
+             Arrays.asList(Collections.singletonList(0), Collections.singletonList(-2), Collections.singletonList(3)),
+             Arrays.asList(null, Collections.singletonList(Integer.MAX_VALUE)),
+             Arrays.asList(Collections.singletonList(5), Arrays.asList(-6, null)),
+             Arrays.asList(Collections.singletonList(Integer.MIN_VALUE), null),
+             null);
+         ColumnVector intExpected = ColumnVector.fromInts(0, -59, 2147483647, -31, -2147483648, 0);
+         ColumnVector intResult = Hash.hiveHash(new ColumnVector[]{nestedIntListCV});) {
+      assertColumnsAreEqual(stringExpected, stringResult);
+      assertColumnsAreEqual(intExpected, intResult);
+    }
+  }
+
+  @Test
+  void testHiveHashNestedType() {
```

Comment:
Better to have a case to verify the case when the max nested depth is larger than 8.